test/test_time.cpp

```diff
@@ -287,6 +287,38 @@ TEST_F(TestTimeFixture, test_rcutils_time_point_value_as_nanoseconds_string) {
   EXPECT_STREQ("-0000000000000000100", buffer);
 }
 
+// Tests the rcutils_time_point_value_as_date_string() function.
+TEST_F(TestTimeFixture, test_rcutils_time_point_value_as_date_string) {
+  rcutils_ret_t ret;
+  rcutils_time_point_value_t timepoint;
+  char buffer[256] = "";
+
+  // Typical use case.
+  timepoint = 100;
+  ret = rcutils_time_point_value_as_date_string(&timepoint, buffer, sizeof(buffer));
+  EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string().str;
+  struct tm tm;
+  ASSERT_TRUE(strptime(buffer, "%Y-%m-%d %H:%M:%S", &tm) != nullptr);
```

Comment:
Does this work on Windows? i guess no?

Reply:
What is the supposed way to support both platforms? Is it to write the code using platform specific instructions and guard with preprocessor directives? Or can we use boost?

Reply:
> Or can we use boost?

We can't use boost, no.  It is too heavy of a dependency.

Given that this is in a test, and we can use C++ there, we can probably do something with https://en.cppreference.com/w/cpp/io/manip/get_time .

Reply:
Ok, the last version uses get_time which should work on Windows as well